### ISSUE TITLE
pgo v4.5.0 standby cluster restoring archive files that do not seem to exist and cluster never initialized but working

### ISSUE THREAD

**Issue by alrooney** [CrunchyData/postgres-operator#1959]
I have a pgo v4.5.0 standby cluster that was created with the following command: 
```
pgo create cluster retroelk-staging \
  --standby \
  --service-type=NodePort \
  --pgbackrest-s3-bucket=<bucket> \
  --pgbackrest-s3-endpoint=s3.amazonaws.com \
  --pgbackrest-s3-region=us-east-1 \
  --pgbackrest-s3-key=<s3-key> \
  --pgbackrest-s3-key-secret=<s3-secret> \
  --username=<username> \
  --password=<password> \
  --password-superuser=<password_superuser> \
  --password-replication=<password_replication> \
  --pgbackrest-storage-type=local,s3 \
  --pgbackrest-repo-path=/backrestrepo/retroelk-backrest-shared-repo \
  --metrics \
  --cpu=1 \
  --memory=1Gi \
  --ccp-image=crunchy-postgres-gis-ha \
  --ccp-image-tag=centos7-11.9-2.4-4.5.0 \
  --pvc-size=50Gi \
  -n pgo
```
It seems to be running and I can connect to the db and do queries but the readiness probe is failing:
```
$ kk get po -n pgo
NAME                                                     READY   STATUS    RESTARTS   AGE
pgo-client-5f59c74967-dlm5d                              1/1     Running   0          3d16h
postgres-operator-6d4f8dc6d5-txl9q                       4/4     Running   0          3d16h
retroelk-staging-59874bbb44-rxtd6                        1/2     Running   0          3d12h
retroelk-staging-backrest-shared-repo-5c946f49f5-m4ppj   1/1     Running   0          3d13h
```
This seems to be because the file "/crunchyadm/pgha_initialized" does not exist.  When I look in the /pgdata/retroelk-staging/pg_log directory at the log I see that it is restoring archive files.  However, these files do not seem to exist in the s3 bucket I have connected this database to?  Where are these files from?  How can I debug/figure out where it is trying to restore these files from and when it might be done/initialized if ever?
```
bash-4.2$ tail -f postgresql-Sun.log
2020-10-11 16:58:14.262 UTC [10741] LOG:  restored log file "000000090000021300000072" from archive
2020-10-11 16:59:14.108 UTC [10741] LOG:  restored log file "000000090000021300000073" from archive
2020-10-11 17:03:14.251 UTC [10741] LOG:  restored log file "000000090000021300000074" from archive
2020-10-11 17:08:14.813 UTC [10741] LOG:  restored log file "000000090000021300000075" from archive
2020-10-11 17:09:14.470 UTC [10741] LOG:  restored log file "000000090000021300000076" from archive
2020-10-11 17:10:14.583 UTC [10741] LOG:  restored log file "000000090000021300000077" from archive
2020-10-11 17:13:14.675 UTC [10741] LOG:  restored log file "000000090000021300000078" from archive
2020-10-11 17:18:14.873 UTC [10741] LOG:  restored log file "000000090000021300000079" from archive
2020-10-11 17:20:16.326 UTC [10741] LOG:  restored log file "00000009000002130000007A" from archive
2020-10-11 17:23:15.130 UTC [10741] LOG:  restored log file "00000009000002130000007B" from archive
```
It has been in this state restoring these files for several days now.

**Comment by jkatz** [comment]
So I may slightly eat my words on #1958 -- the readiness probe should be :+1: even if it's in a standby mode. @andrewlecuyer -- thoughts?

Outside of that, this is the correct behavior: a "standby" keeps replaying WAL logs from an archive until it's promoted. It does not have a direct replication connection to the Postgres primary, as this is designed to work across multiple data centers and federation across Kubernetes clusters is still in the nascent stages. However, replaying through WAL archives is an appropriate way to keep a standby up-to-date -- this was actually the method used before streaming replication was introduced in Postgres 9.0 :smile:

**Comment by andrewlecuyer** [comment]
@alrooney thanks for bringing this to our attention.

It turns out there was a behavior change in Patroni 2.0.0 (this version of Patroni was introduced in PostgreSQL Operator v4.5.0), in which the `master` endpoint we were previously using to check whether or not a local instance is running as the leader of a standby cluster no longer works.  This is specifically the result of the following Patroni bug fix, as indicated in the [ Patroni 2.0.0 release notes](https://patroni.readthedocs.io/en/latest/releases.html#version-2-0-0):

> The GET /standby-leader for a master and GET /master for a standby_leader were incorrectly responding with 200.

Instead we now need to leverage the `standby_leader` endpoint when initializing a standby cluster, and the following patch has been submitted to address this:

https://github.com/CrunchyData/crunchy-containers/pull/1285

To workaround this issue in the meantime, once your standby primary is healthy as indicated by a `running` status when running the `patronictl list` command inside of the container, e.g.:

```bash
$ kubectl exec -it standbycluster-6d87d94886-wfb8r -n pgouser1 -- patronictl list
+ Cluster: standbycluster (6882485012892790929) ---+----------------+---------+----+-----------+
| Member                         | Host           | Role           | State   | TL | Lag in MB |
+--------------------------------+----------------+----------------+---------+----+-----------+
| standbycluster-6d87d94886-wfb8r | 10.244.196.176 | Standby Leader | running |  1 |           |
+--------------------------------+----------------+----------------+---------+----+-----------+
```

You can then simply add the proper `/crunchyadm/pgha_initialized` file:

```bash
$ kubectl exec -it standbycluster-6d87d94886-wfb8r -n pgouser1 -- touch "/crunchyadm/pgha_initialized"
```

From here the next readiness check should succeed, and the standby cluster should proceed with initialization as expected.

**Comment by alrooney** [comment]
@andrewlecuyer thanks!  That worked great.
